src/eth_provider/database/types/receipt.rs

```diff
@@ -58,7 +55,7 @@ impl<'a> arbitrary::Arbitrary<'a> for StoredTransactionReceipt {
                 logs_bloom: Bloom::arbitrary(u)?,
                 state_root: Some(B256::arbitrary(u)?),
                 status_code: Some(U64::from(receipt.success as u8)),
-                transaction_type: U8::from::<u8>(receipt.tx_type.into()),
+                transaction_type: U8::from::<u8>(Into::<u8>::into(receipt.tx_type) % 3),
```

Comment:
why do we need this? Isn't `receipt.tx_type` already a transaction type?

Reply:
`receipt.tx_type` is of type `TxType` but in our RPC we don't support `Eip4844` instance (number 3). So I first transform the `TxType` into a `u8`, then I do a mod 3 to avoid potential `Eip4844` type and finally I convert to `U8` which is the type required by the rpc `transaction_type` type.